Source/Core/DolphinQt2/MainWindow.cpp

```diff
@@ -480,3 +518,17 @@ void MainWindow::SetStateSlot(int slot)
   Settings::Instance().SetStateSlot(slot);
   m_state_slot = slot;
 }
+
+bool MainWindow::eventFilter(QObject* object, QEvent* event)
+{
+  if (event->type() == QEvent::Close)
```

Comment:
You can get rid of an extra level of indentation by combining the conditions: `if (event->type() == QEvent::Close && !Stop())`